hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobUtils.java

```diff
@@ -907,6 +789,143 @@ public static boolean hasMobColumns(TableDescriptor htd) {
+                sf.initReader();
+                byte[] mobRefData = sf.getMetadataValue(HStoreFile.MOB_FILE_REFS);
+                byte[] mobCellCountData = sf.getMetadataValue(HStoreFile.MOB_CELLS_COUNT);
+                byte[] bulkloadMarkerData = sf.getMetadataValue(HStoreFile.BULKLOAD_TASK_KEY);
+                if (mobRefData == null && (mobCellCountData != null ||
+                    bulkloadMarkerData == null)) {
+                  LOG.info("Found old store file with no MOB_FILE_REFS: " + pp
+                    +" - can not proceed until all old files will be MOB-compacted");
+                  return;
+                } else if (mobRefData == null) {
+                  LOG.info("Skipping file without MOB references (can be bulkloaded file):"+ pp);
+                  continue;
+                }
+                String[] mobs = new String(mobRefData).split(",");
+                regionMobs.addAll(Arrays.asList(mobs));
+              }
+            } catch (FileNotFoundException e) {
+              //TODO
+              LOG.warn(e.getMessage());
+              continue;
+            }
+            succeed = true;
+          }
+          // Add MOB refs for current region/family
+          allActiveMobFileName.addAll(regionMobs);
+        } // END column families
+      }//END regions
+
+      // Now scan MOB directories and find MOB files with no references to them
+      long now = System.currentTimeMillis();
+      long minAgeToArchive = conf.getLong(MobConstants.MOB_MINIMUM_FILE_AGE_TO_ARCHIVE_KEY,
+                              MobConstants.DEFAULT_MOB_MINIMUM_FILE_AGE_TO_ARCHIVE);
+      for (ColumnFamilyDescriptor hcd: list) {
+          List<Path> toArchive = new ArrayList<Path>();
+          String family = hcd.getNameAsString();
+          Path dir = getMobFamilyPath(conf, table, family);
+          RemoteIterator<LocatedFileStatus> rit = fs.listLocatedStatus(dir);
+          while(rit.hasNext()) {
+            LocatedFileStatus lfs = rit.next();
+            Path p = lfs.getPath();
+            if (!allActiveMobFileName.contains(p.getName())) {
+              // MOB is not in a list of active references, but it can be too
+              // fresh, skip it in this case
+              /*DEBUG*/ LOG.debug(" Age=" + (now - fs.getFileStatus(p).getModificationTime()) +
+                " MOB file="+ p);
+              if (now - fs.getFileStatus(p).getModificationTime() > minAgeToArchive) {
+                toArchive.add(p);
+              } else {
+                LOG.debug(" Skipping fresh file: " + p);
+              }
+            }
+          }
+          LOG.info(" MOB Cleaner found "+ toArchive.size()+" files for family="+family);
+          removeMobFiles(conf, table, family.getBytes(), toArchive);
+          LOG.info(" MOB Cleaner archived "+ toArchive.size()+" files");
+      }
+    }
+  }
+
+
+
+
+  public static long getNumberOfMobFiles(Configuration conf, TableName tableName, String family)
```

Comment:
nit: This is only needed for test? Can we keep it in a test file?

Reply:
Fixed.